controllers/kubeletconfig_controller.go

```diff
@@ -275,6 +295,51 @@ func (r *KubeletConfigReconciler) makeKCHandlerForPlatform(ctx context.Context,
 	return nil, fmt.Errorf("unsupported platform: %s", r.Platform)
 }
 
+func (r *KubeletConfigReconciler) deleteConfigMap(ctx context.Context, instance *nropv1.NUMAResourcesOperator, kcKey client.ObjectKey) (*corev1.ConfigMap, bool, error) {
+	cm := getDeletedOwner(kcKey, r.GarbageCollectionFor)
+	if cm == nil {
+		return nil, false, nil
+	}
+	// we'll get to this flow only on hypershift
+	// on openshift the deletion is done automatically by setting the owner reference on the dependent ConfigMap
+	nodePoolName := cm.Labels[HyperShiftNodePoolLabel]
+	generatedName := objectnames.GetComponentName(instance.Name, nodePoolName)
+	dependentCM := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generatedName,
+			Namespace: r.Namespace,
+		},
+	}
+	if err := r.Delete(ctx, dependentCM); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return cm, true, err
+		}
+		klog.V(2).InfoS("could not delete ConfigMap since it was not found", "configmapName", dependentCM.Name)
+	}
+	r.GarbageCollectionFor = removeDeletedOwner(kcKey, r.GarbageCollectionFor)
+	return cm, true, nil
+}
+
+func getDeletedOwner(kcKey client.ObjectKey, ownerConfigMaps []corev1.ConfigMap) *corev1.ConfigMap {
+	for i := range ownerConfigMaps {
+		cm := &ownerConfigMaps[i]
+		if client.ObjectKeyFromObject(cm).String() == kcKey.String() {
```

Comment:
let's check if there's a recommended way to compare keys. I don't see obvious issues with this one, but still